L1Trigger/Phase2L1Taus/plugins/HPSPFTauProducer.cc

```diff
@@ -0,0 +1,317 @@
+#include "L1Trigger/Phase2L1Taus/plugins/HPSPFTauProducer.h"
+#include "FWCore/Utilities/interface/InputTag.h"
+#include <cmath>  // std::fabs
+
+HPSPFTauProducer::HPSPFTauProducer(const edm::ParameterSet& cfg)
+    : moduleLabel_(cfg.getParameter<std::string>("@module_label")),
+      tauBuilder_(cfg),
+      useChargedPFCandSeeds_(cfg.getParameter<bool>("useChargedPFCandSeeds")),
+      minSeedChargedPFCandPt_(cfg.getParameter<double>("minSeedChargedPFCandPt")),
+      maxSeedChargedPFCandEta_(cfg.getParameter<double>("maxSeedChargedPFCandEta")),
+      maxSeedChargedPFCandDz_(cfg.getParameter<double>("maxSeedChargedPFCandDz")),
+      useJetSeeds_(cfg.getParameter<bool>("useJetSeeds")),
+      minSeedJetPt_(cfg.getParameter<double>("minSeedJetPt")),
+      maxSeedJetEta_(cfg.getParameter<double>("maxSeedJetEta")),
+      minPFTauPt_(cfg.getParameter<double>("minPFTauPt")),
+      maxPFTauEta_(cfg.getParameter<double>("maxPFTauEta")),
+      minLeadChargedPFCandPt_(cfg.getParameter<double>("minLeadChargedPFCandPt")),
+      maxLeadChargedPFCandEta_(cfg.getParameter<double>("maxLeadChargedPFCandEta")),
+      maxLeadChargedPFCandDz_(cfg.getParameter<double>("maxLeadChargedPFCandDz")),
+      maxChargedIso_(cfg.getParameter<double>("maxChargedIso")),
+      maxChargedRelIso_(cfg.getParameter<double>("maxChargedRelIso")),
+      deltaRCleaning_(cfg.getParameter<double>("deltaRCleaning")),
+      applyPreselection_(cfg.getParameter<bool>("applyPreselection")),
+      debug_(cfg.getUntrackedParameter<bool>("debug", false)) {
+  if (debug_) {
+    std::cout << "<HPSPFTauProducer::HPSPFTauProducer (moduleLabel = " << moduleLabel_ << ")>:" << std::endl;
+  }
+
+  srcL1PFCands_ = cfg.getParameter<edm::InputTag>("srcL1PFCands");
+  tokenL1PFCands_ = consumes<l1t::PFCandidateCollection>(srcL1PFCands_);
+  srcL1Jets_ = cfg.getParameter<edm::InputTag>("srcL1Jets");
+  if (useJetSeeds_) {
+    tokenL1Jets_ = consumes<std::vector<reco::CaloJet>>(srcL1Jets_);
+  }
+  srcL1Vertices_ = cfg.getParameter<edm::InputTag>("srcL1Vertices");
+  if (srcL1Vertices_.label() != "") {
+    tokenL1Vertices_ = consumes<std::vector<l1t::TkPrimaryVertex>>(srcL1Vertices_);
+  }
+  deltaR2Cleaning_ = deltaRCleaning_ * deltaRCleaning_;
+
+  edm::ParameterSet cfg_signalQualityCuts = cfg.getParameter<edm::ParameterSet>("signalQualityCuts");
+  signalQualityCutsDzCutDisabled_ = readL1PFTauQualityCuts(cfg_signalQualityCuts, "disabled");
+  edm::ParameterSet cfg_isolationQualityCuts = cfg.getParameter<edm::ParameterSet>("isolationQualityCuts");
+  isolationQualityCutsDzCutDisabled_ = readL1PFTauQualityCuts(cfg_isolationQualityCuts, "disabled");
+
+  produces<l1t::HPSPFTauCollection>();
+}
+
+namespace {
```

Comment:
more an aside, due to overloading these two functions could have had the same name, this would be the preferred way

Reply:
Tried but it didn't work